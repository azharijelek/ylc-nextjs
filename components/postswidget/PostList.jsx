import Link from 'next/link'

/**
 * PostList
 * @param {*} props id, title, thumbnail, permalink
 */
export default function PostList(props) {
  return (
    <>
      <article id={'post-' + props.id} className={'post post-' + props.id}>
        <div className="postThumb">
          <div className="media">
            <Link href={props.permalink}>
              <a>
                <img
                  src={props.thumbnail}
                  alt={props.title}
                  width="60"
                  height="60"
                  loading="lazy"
                />
              </a>
            </Link>
          </div>
        </div>
        <div className="postDetail">
          <h4>
            <Link href={props.permalink}>
              <a title={props.title}>{props.title}</a>
            </Link>
          </h4>
        </div>
      </article>

      <style jsx>{`
        .post {
          display: flex;
          margin-bottom: 20px;
        }
        .postThumb {
          min-width: 85px;
          max-width: 85px;
          margin-right: 15px;
        }
        .postDetail {
          h4 {
            margin: 0;
            font-size: 16px;
            line-height: 20px;
          }
        }
        .media {
          width: 100%;
          height: 60px;
          overflow: hidden;
          border-radius: 4px;
          background: #e0e0e0;
          img {
            object-fit: cover;
            height: 100%;
            width: 100%;
          }
        }
      `}</style>
    </>
  )
}
