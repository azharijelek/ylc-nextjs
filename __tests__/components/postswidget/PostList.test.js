import React from 'react'
import { shallow } from 'enzyme'
import PostList from '@/components/postswidget/PostList'

describe('<PostList/>', () => {
  let wrapper, post, PostListComp

  beforeEach(() => {
    post = {
      featured_img:
        'https://cdn.statically.io/img/yourlifechoices.s3-ap-southeast-2.amazonaws.com/old/assets/autouploads/f4aa5f507148.jpg?f=webp',
      id: 533593,
      permalink:
        'http://localhost:4200/technology/safety-online/secure-access/password-managers-explained/',
      slug: '/technology/safety-online/secure-access/password-managers-explained/',
      title: 'Password managers explained: do you need one? Are they safe?'
    }

    PostListComp = (
      <PostList
        id={post.id}
        title={post.title}
        thumbnail={post.featured_img}
        permalink={post.permalink}
        key={'PopularPost-' + post.id}
      />
    )

    wrapper = shallow(PostListComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have article with class: .post-${post_id}', () => {
    expect(wrapper.find('article')).toHaveLength(1)
    expect(wrapper.find('.post-' + post.id)).toHaveLength(1)
  })

  it('should have postThumb', () => {
    expect(wrapper.find('.postThumb')).toHaveLength(1)
    expect(wrapper.find('.postThumb').find('.media')).toHaveLength(1)
    expect(wrapper.find('.media').find('img')).toHaveLength(1)
  })

  it('should render correct post thumbnail structure', () => {
    const img = wrapper.find('.media').find('img')
    expect(img.prop('src')).toBe(post.featured_img)
    expect(img.prop('alt')).toBe(post.title)
    expect(img.prop('width')).toBe('60')
    expect(img.prop('height')).toBe('60')
    expect(img.prop('loading')).toBe('lazy')
  })

  it('should have postDetail', () => {
    expect(wrapper.find('.postDetail')).toHaveLength(1)
  })

  it('should have correct postDetail elements', () => {
    expect(wrapper.find('.postDetail').find('h4')).toHaveLength(1)
    expect(wrapper.find('.postDetail').find('h4 Link')).toHaveLength(1)
    expect(wrapper.find('.postDetail').find('h4 Link a')).toHaveLength(1)
  })

  it('should render correct permalink', () => {
    expect(wrapper.find('.postDetail').find('h4 Link').prop('href')).toBe(post.permalink)
    expect(wrapper.find('.postDetail').find('h4 Link a').prop('title')).toBe(post.title)
  })
})
