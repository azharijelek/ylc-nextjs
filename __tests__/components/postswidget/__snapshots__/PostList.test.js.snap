// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostList/> should renders correctly 1`] = `ShallowWrapper {}`;
